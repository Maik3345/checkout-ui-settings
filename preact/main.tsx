// Import necessary functions and types
import { getOrderForm, initializeApp } from "@/shared";

// Import global CSS styles
import pkg from "../package.json";
import "./src/styles/index.css";
import "./src/styles/reset.css";

// Define global application name
// Import package.json to access the project name

// Define global application name using the name from package.json
const APP_NAME = pkg.name;
const APP_VERSION = pkg.version;

// Log welcome message to console
console.log(`üéâ Yay! You are using the ${APP_NAME} version ${APP_VERSION}!!`);

try {
  // Fetch the order form and render the application
  // when the data is available
  getOrderForm((orderForm: VtexOrderForm | undefined) => initializeApp(orderForm));
} catch (error) {
  // Handle any errors that occur during initialization
  console.error(
    `‚ùå Error initializing ${APP_NAME} version ${APP_VERSION}: `,
    error
  );
}
