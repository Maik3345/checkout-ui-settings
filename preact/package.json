{
  "name": "vite-preact-ts",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "build:watch": "tsc && vite build --watch --emptyOutDir",
    "preview": "vite preview",
    "test": "jest --clearCache && jest",
    "test:watch": "jest --clearCache && jest --watch",
    "lint": "eslint ./src --ext .ts,.tsx,.js,.jsx"
  },
  "dependencies": {
    "preact": "^10.16.0"
  },
  "devDependencies": {
    "@preact/preset-vite": "^2.5.0",
    "@testing-library/preact": "^3.2.3",
    "@types/jest": "^29.5.7",
    "@types/jquery": "^3.5.25",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "prettier": "^3.0.3",
    "ts-jest": "^29.1.1",
    "typescript": "^5.2.2",
    "vite": "^4.4.8"
  }
}
