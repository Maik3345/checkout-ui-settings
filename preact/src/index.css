body {
    margin: 0;
    font-family: Arial, sans-serif;
    background-color: #f4f4f4;
}

.container {
    max-width: 800px;
    margin: 0 auto;
    padding: 20px;
}