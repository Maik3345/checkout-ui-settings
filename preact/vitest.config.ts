/// <reference types="vitest" />
import { defineConfig } from 'vitest/config';
import { resolve } from 'path';

export default defineConfig({
  test: {
    globals: true,
    environment: 'happy-dom',
    setupFiles: './src/test/setup.ts',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      provider: 'v8',
      reporter: ['text', 'json', 'html'],
      reportsDirectory: './coverage',
      include: ['src/**/*.{js,jsx,ts,tsx}'],
      exclude: ['**/*.test.*', '**/node_modules/**', 'src/test/**'],
    },
  },
  resolve: {
    alias: {
      '@': resolve(__dirname, './src'),
    },
  },
});
