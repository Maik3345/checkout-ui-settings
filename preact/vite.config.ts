import path from 'node:path';
import preact from '@preact/preset-vite';
import { defineConfig } from 'vite';

export default defineConfig({
  resolve: {
    alias: {
      '@': path.resolve(__dirname, './src'),
    },
  },
  plugins: [preact()],
  build: {
    outDir: '../checkout-ui-custom',
    assetsDir: '',
    sourcemap: true,
    rollupOptions: {
      input: 'src/main.tsx',
      output: {
        format: 'iife',
        name: 'myApp',
        entryFileNames: 'checkout6-custom.js',
        chunkFileNames: 'chunks/[name]-custom.js',
        assetFileNames: 'checkout6-custom.css',
      },
    },
    // Added target configuration for Vite 6
    target: 'es2015',
  },
});
