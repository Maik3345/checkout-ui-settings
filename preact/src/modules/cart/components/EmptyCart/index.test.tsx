import { describe, it, expect } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/preact';
import { EmptyCart } from './index';

describe('EmptyCart', () => {
  it('should render correctly with initial state', () => {
    render(<EmptyCart />);
    
    // Verificar que el título se muestra correctamente
    const titleElement = screen.getByText(/I am the custom empty cart/, { exact: false });
    expect(titleElement).toBeTruthy();
    
    // Verificar que el botón está presente
    const buttonElement = screen.getByRole('button', { name: /Click me/ });
    expect(buttonElement).toBeTruthy();
    
    // Verificamos que el componente contenga el texto del contador
    const h2Element = screen.getByRole('heading', { level: 2 });
    expect(h2Element.textContent).toContain('0');
  });

  it('should increment counter when the button is clicked', async () => {
    render(<EmptyCart />);
    
    // Obtiene el botón
    const button = screen.getByRole('button', { name: /Click me/ });
    
    // Simula un clic en el botón
    fireEvent.click(button);
    
    // Verifica que el contador incrementó a 1
    const h2Element = screen.getByRole('heading', { level: 2 });
    expect(h2Element.textContent).toContain('1');
    
    // Simula otro clic
    fireEvent.click(button);
    
    // Verifica que el contador incrementó a 2
    expect(h2Element.textContent).toContain('2');
  });

  it('should have the correct CSS class', () => {
    const { container } = render(<EmptyCart />);
    
    // Verifica que el contenedor tiene la clase CSS correcta
    const containerElement = container.firstChild as HTMLElement;
    expect(containerElement.className).toContain('notFoundContainer');
  });

  it('should match snapshot', () => {
    const { container } = render(<EmptyCart />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
