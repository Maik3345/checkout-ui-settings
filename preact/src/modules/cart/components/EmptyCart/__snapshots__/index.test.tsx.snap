// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyCart > should match snapshot 1`] = `
<div
  class="_notFoundContainer_a509b7"
>
  <h2>
    I am the custom empty cart
    <button>
      Click me
    </button>
    0
  </h2>
</div>
`;
